src/Mvc/src/Microsoft.AspNetCore.Mvc.Core/BindPropertyAttribute.cs

```diff
@@ -2,31 +2,102 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
+using System.ComponentModel;
 using Microsoft.AspNetCore.Mvc.ModelBinding;
 
 namespace Microsoft.AspNetCore.Mvc
 {
+    /// <summary>
+    /// An attribute that can specify a model name or type of <see cref="IModelBinder"/> to use for binding the
+    /// associated property.
+    /// </summary>
+    /// <remarks>
+    /// Similar to <see cref="ModelBinderAttribute"/>. Unlike that attribute, <see cref="BindPropertyAttribute"/>
+    /// applies only to properties and adds an <see cref="IRequestPredicateProvider"/> implementation that by default
+    /// indicates the property should not be bound for HTTP GET requests (see also <see cref="SupportsGet"/>).
+    /// </remarks>
     [AttributeUsage(AttributeTargets.Property, AllowMultiple = false, Inherited = true)]
     public class BindPropertyAttribute : Attribute, IModelNameProvider, IBinderTypeProviderMetadata, IRequestPredicateProvider
     {
+        private static readonly BindingSource[] BindingSources = new[]
+        {
+            BindingSource.Body,
+            BindingSource.Custom,
+            BindingSource.Form,
+            BindingSource.FormFile,
+            BindingSource.Header,
+            BindingSource.ModelBinding,
+            BindingSource.Path,
+            BindingSource.Query,
+            BindingSource.Services,
+            BindingSource.Special,
+        };
+
         private static readonly Func<ActionContext, bool> _supportsAllRequests = (c) => true;
 
         private static readonly Func<ActionContext, bool> _supportsNonGetRequests = IsNonGetRequest;
 
         private BindingSource _bindingSource;
 
+        /// <summary>
+        /// Initializes a new instance of <see cref="BindPropertyAttribute"/>.
+        /// </summary>
+        /// <remarks>
+        /// If setting <see cref="BinderType"/> to an <see cref="IModelBinder"/> implementation that does not use values
+        /// from form data, route values or the query string, instead use the
+        /// <see cref="BindPropertyAttribute(BindingSourceKey)"/> constructor to set <see cref="BindingSource"/>.
+        /// </remarks>
+        public BindPropertyAttribute()
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="BindPropertyAttribute"/>.
+        /// </summary>
+        /// <param name="bindingSource">
+        /// The <see cref="BindingSourceKey"/> that indicates the value of the
+        /// <see cref="BindingSource"/> property.
+        /// </param>
+        public BindPropertyAttribute(BindingSourceKey bindingSource)
```

Comment:
Why add this? `[FromForm]` does the needful on its own. I don't know why we'd introduce more ways to do the same thing, and it's pretty rare to use a binding source in razor pages.